### DISCUSSION TITLE
`$lookup` and `$validate-code` operations return "not found"/false for valid CPT codes in local CodeSystem

### DISCUSSION THREAD

**Discussion by dylman123** [medplum/medplum#5573] [Q&A]
## Description
Hey Medplum team, I'm encountering issues with the terminology operations `$lookup` and `$validate-code` when trying to query codes from a locally stored CPT CodeSystem. Both operations are failing despite the code being present in the CodeSystem.

## Environment
- medplum-server version: 3.2.23
- Running locally
- Using CPT CodeSystem

## Steps to Reproduce
1. Created a CPT CodeSystem with code "0427U" (and others)
2. Verified CodeSystem is accessible via GET request to `/CodeSystem/{id}`
3. Attempted `$lookup` operation:
   ```
   GET /fhir/R4/CodeSystem/$lookup?system=http://www.ama-assn.org/go/cpt&code=0427U
   ```
4. Attempted `$validate-code` operation:
   ```
   GET /fhir/R4/CodeSystem/$validate-code?url=http://www.ama-assn.org/go/cpt&code=0427U
   ```

## Current Behavior
- `$lookup` returns a 404 "Not found" error
- `$validate-code` returns `{ "resourceType": "Parameters", "parameter": [{ "name": "result", "valueBoolean": false }] }`

## Expected Behavior
Both operations should successfully find and validate the code since it exists in the CodeSystem.

This is blocking my implementation of terminology validation. Any guidance would be appreciated!

## Additional Context
I've attached screenshots showing:
1. The CodeSystem resource [content](https://github.com/user-attachments/files/17852549/code-system.json)
2. The $lookup operation response
3. The $validate-code operation response

![Screenshot 2024-11-21 at 4 49 02 PM](https://github.com/user-attachments/assets/aa50b194-6a2d-46d3-83c0-9c8a334c3cd3)

![Screenshot 2024-11-21 at 4 49 26 PM](https://github.com/user-attachments/assets/38ceaec7-110a-4f34-bce5-67e7bbed5570)

**Comment by codyebberson** [comment]
Hi @dylman123  - 

By default, the Medplum terminology service only includes the CodeSystem/ValueSets that are included in the FHIR R4 spec, which is a very tiny subset of all codes.

Medplum can import all of the UMLS CodeSystem/ValueSets (SNOMED, CPT, ICD, RxNorm, etc), but there are some legal and licensing challenges about doing that automatically.

You can see some of that discussion here: https://github.com/medplum/medplum/discussions/1411

Please contact us directly if you'd like to discuss more:  hello@medplum.com
